concent_api/common/decorators.py

```diff
@@ -169,9 +171,45 @@ def wrapper(request, client_message, client_public_key, *args, **kwargs):
             try:
                 if database_name is not None:
                     sid = transaction.savepoint(using=database_name)
-                response_from_view = view(request, client_message, client_public_key, *args, **kwargs)
+                try:
+                    response_from_view = view(request, client_message, client_public_key, *args, **kwargs)
+                except DatabaseError as e:
```

Comment:
whole handling should be extracted to a separate function

Reply:
done